test/fhirtypes/ElementDefinition.assignString.test.ts

```diff
@@ -1175,4 +1176,21 @@ describe('ElementDefinition', () => {
       });
     });
   });
+  describe('#checkXhtml', () => {
+    beforeEach(() => {
+      loggerSpy.reset();
+    });
+    it("should warn and throw when path doesn't start and end with div", () => {
+      const narrativeDiv = patient.findElementByPath('text.div', fisher);
+      expect(() =>
+        narrativeDiv.assignValue('<piv xmlns="http://www.w3.org/1999/xhtml">Twas brillig</div>')
+      ).toThrow();
+    });
```

Comment:
Since the test description says that it should warn, the test should check the logger for a warning message.

Reply:
Julian and I actually went over this, we found that the logger is being called in that test (we can see that it's being tested in the coverage report), but because that specific case in ElementDefinition is designed to throw an exception, we can't actually check the logger in the test case itself. ( See `throw new MismatchedTypeError('string', value, type);` in function `assignString()` ) Is there a way to test the log when the program crashes? Not sure if there's an another test case out there that addresses this specific kind of situation.

Reply:
I assumed that `expect(...).toThrow()` would catch the exception so you could check the log on the next line (after the line w/ `).toThrow();`.  Is that not the case?

If not, another common approach is to use a `try`/`catch` inside the test and put any additional assertions in the `catch`.  E.g.
```ts
let didThrow = false;
const narrativeDiv = patient.findElementByPath('text.div', fisher);
try {
  narrativeDiv.assignValue('<piv xmlns="http://www.w3.org/1999/xhtml">Twas brillig</div>');
} catch (error) {
  didThrow = true;
  // Check error here to see if it is the error you expect
  // Check loggerSpy here to see if it logged the message you expect
}
expect(didThrow).toBeTrue();
```
_NOTE: The code above is off the cuff.  I didn't try to compile it, so it might not be exactly right, but you get the idea._

Reply:
@cmoesel Thank you for the help, this worked!